Client/Frontend/Widgets/BlurTableViewController.swift

```diff
@@ -0,0 +1,182 @@
+        return BlurTableViewAction(title: NSLocalizedString("Delete from History", comment: "Context Menu Action for Activity Stream"), iconString: "action_delete", handler: { action in
+            self.profile.history.removeHistoryForURL(self.site.url)
+        })
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.actions = [bookmarkAction, deleteFromHistoryAction]
+        view.backgroundColor = UIColor.clearColor().colorWithAlphaComponent(0.4)
+
+        view.addGestureRecognizer(tapRecognizer)
+
+        view.addSubview(tableView)
+        tableView.snp_makeConstraints { make in
+            make.center.equalTo(self.view)
+            make.width.equalTo(290)
+            make.height.equalTo(73 + actions.count * 56)
+        }
+
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.keyboardDismissMode = UIScrollViewKeyboardDismissMode.OnDrag
+        tableView.backgroundColor = UIConstants.PanelBackgroundColor
+        tableView.scrollEnabled = false
+        tableView.separatorColor = UIConstants.SeparatorColor
+        tableView.layer.cornerRadius = 10
+        tableView.registerClass(BlurTableViewCell.self, forCellReuseIdentifier: "BlurTableViewCell")
+        tableView.registerClass(BlurTableViewHeaderCell.self, forCellReuseIdentifier: "BlurTableViewHeaderCell")
+
+        if #available(iOS 9, *) {
+            tableView.cellLayoutMarginsFollowReadableWidth = false
+        }
+    }
+
+    func dismiss(gestureRecognizer: UIGestureRecognizer) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+    }
+
+    deinit {
+        // The view might outlive this view controller thanks to animations;
+        // explicitly nil out its references to us to avoid crashes. Bug 1218826.
+        tableView.dataSource = nil
+        tableView.delegate = nil
+    }
+
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return actions.count + 1
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        if indexPath.row == 0 {
+            return
+        }
+        let action = actions[indexPath.row - 1]
+        return action.handler(action)
+    }
+
+    func tableView(tableView: UITableView, hasFullWidthSeparatorForRowAtIndexPath indexPath: NSIndexPath) -> Bool {
+        return true
+    }
+
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return indexPath.row == 0 ? 74 : 56
+    }
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        if indexPath.row == 0 {
+            let cell = tableView.dequeueReusableCellWithIdentifier("BlurTableViewHeaderCell", forIndexPath: indexPath) as! BlurTableViewHeaderCell
```

Comment:
Have you thought about just using a UItableview header here instead? It's called a header here but it is actually just a cell. If you changed it, You could avoid having to configure it here like this.